src/reads-stdin.c

```diff
@@ -0,0 +1,68 @@
+#include <stdio.h>
+#include <string.h>
+
+// This program reads stdin and asserts that it is the expected value repeated
+// until stdin ends
+
+// Intentionally a prime length to make hitting buffering edge cases more likely
+const char expected_repeat[] = {'e', 'x', 'a', 'm', 'p', 'l', 'e'};
+
+int main(int argc, char** argv)
+{
+    size_t consumed_prefix = 0;
+    char buffer[20];
+    for (;;)
+    {
+        size_t read_amount = fread(buffer, 1, sizeof(buffer), stdin);
+        if (argc > 1)
+        {
+            puts(argv[1]);
+        }
+
+        if (read_amount == 0)
+        {
+            if (feof(stdin))
+            {
+                puts("success");
+                return 0;
+            }
+
+            return 1;
+        }
+
+        size_t prefix_check = read_amount;
+        if (prefix_check <= (sizeof(expected_repeat) - consumed_prefix))
```

Comment:
```suggestion
        if (prefix_check < (sizeof(expected_repeat) - consumed_prefix))
```
Otherwise, if `prefix_check` is 7 and `consumed_prefix` is 0, this will result `consumed_prefix` of 7 and a `prefix_check` below of 0 (which seems to be an invariant violation).